packages/react-examples/src/react/DetailsList/DetailsList.CustomGroupHeaders.Example.tsx

```diff
@@ -46,35 +49,64 @@ const classNames = mergeStyleSets({
 const ITEMS_PER_GROUP = 20;
 const GROUP_COUNT = 20;
 
-export class DetailsListCustomGroupHeadersExample extends React.Component<{}, {}> {
+export class DetailsListCustomGroupHeadersExample extends React.Component<{}, { selectedItem: IDropdownOption }> {
   private _items: IExampleItem[];
   private _groups: IGroup[];
+  private dropdownStyles: Partial<IDropdownStyles> = { dropdown: { width: 300 } };
+  private dropdownControlledExampleOptions = [
+    { key: 'onHover', text: 'CheckboxVisibility.onHover' },
+    { key: 'always', text: 'CheckboxVisibility.always' },
+    { key: 'hidden', text: 'CheckboxVisibility.hidden' },
+  ];
 
   constructor(props: {}) {
     super(props);
 
     this._items = createListItems(500);
     this._groups = createGroups(GROUP_COUNT, 1, 0, ITEMS_PER_GROUP);
+    this.state = { selectedItem: this.dropdownControlledExampleOptions[0] };
   }
 
   public render(): JSX.Element {
     return (
-      <DetailsList
-        items={this._items}
-        groups={this._groups}
-        groupProps={{
-          onRenderHeader: this._onRenderGroupHeader,
-          onRenderFooter: this._onRenderGroupFooter,
-        }}
-        getGroupHeight={this._getGroupHeight}
-        ariaLabelForSelectionColumn="Toggle selection"
-        ariaLabelForSelectAllCheckbox="Toggle selection for all items"
-        checkButtonAriaLabel="select row"
-        onRenderDetailsHeader={this._onRenderDetailsHeader}
-      />
+      <>
+        <Dropdown
+          label="Checkbox visibility"
+          selectedKey={this.state.selectedItem ? this.state.selectedItem.key : undefined}
+          // eslint-disable-next-line react/jsx-no-bind
+          onChange={this._onChange}
+          placeholder="Select checkbox visibility"
+          options={this.dropdownControlledExampleOptions}
+          styles={this.dropdownStyles}
+        />
+        <DetailsList
+          items={this._items}
+          groups={this._groups}
+          groupProps={{
+            onRenderHeader: this._onRenderGroupHeader,
+            onRenderFooter: this._onRenderGroupFooter,
+          }}
+          getGroupHeight={this._getGroupHeight}
+          ariaLabelForSelectionColumn="Toggle selection"
+          ariaLabelForSelectAllCheckbox="Toggle selection for all items"
+          checkButtonAriaLabel="select row"
+          onRenderDetailsHeader={this._onRenderDetailsHeader}
+          checkboxVisibility={this._getCheckboxVisibilityForSelectedKey(this.state.selectedItem)}
```

Comment:
I think you should be able to just do this instead of having a separate method:
```suggestion
          checkboxVisibility={CheckboxVisibility[this.state.selectedItem.key as keyof typeof CheckboxVisibility]}
```

Reply:
Done!